rechunker/api.py

```diff
@@ -579,3 +585,45 @@ def _setup_array_rechunk(
     int_proxy = ArrayProxy(int_array, int_chunks)
     write_proxy = ArrayProxy(target_array, write_chunks)
     return CopySpec(read_proxy, int_proxy, write_proxy)
+
+
+@contextlib.contextmanager
+def rechunk_cm(
```

Comment:
It might be nice if we could just call `rechunk` directly as a context manager and get the right behavior, rather than introducing a new function to the API.

Reply:
Done, I rename `rechunk` to `_unsafe_rechunk` and `rechunk_cm` to `rechunk`. Only the later is exposed in __init__